api/audit/models.py

```diff
@@ -93,6 +111,34 @@ def update_environments_updated_at(self):
             updated_at=self.created_date
         )
 
+    @hook(
+        AFTER_CREATE,
+        priority=priority.HIGH_PRIORITY,
+        when="environment_document_updated",
+        is_now=True,
+    )
+    def send_environments_to_dynamodb(self):
+        from environments.models import Environment
+
+        Environment.write_environments_to_dynamodb(
+            environment_id=self.environment_id, project_id=self.project_id
+        )
+
+    @hook(
+        AFTER_CREATE,
+        when="environment_document_updated",
+        is_now=True,
+    )
+    def send_environment_update_message(self):
```

Comment:
So, as I understand it, this is all happening in the same thread when an AuditLog gets written. So a task runner, when it creates an AuditLog will have to: 

 1. Create the AuditLogRecord
 2. Update any associated environments in the db with the new `updated_at` value
 3. Write the updated document to dynamodb
 4. Send the update message to fastly for real time

I'm not sure if this is an issue or not, but it is a _little_ difficult to follow. Perhaps it's something that we can refactor later?

Reply:
> So, as I understand it, this is all happening in the same thread when an AuditLog gets written. So a task runner, when it creates an AuditLog will have to:
> 
>     1. Create the AuditLogRecord
> 
>     2. Update any associated environments in the db with the new `updated_at` value
> 
>     3. Write the updated document to dynamodb
> 
>     4. Send the update message to fastly for real time
> 
> 
> I'm not sure if this is an issue or not, but it is a _little_ difficult to follow. Perhaps it's something that we can refactor later?

It's mostly correct except the 4th point - it creates a task to send the SSE message